src/activity/service.ts

```diff
@@ -0,0 +1,76 @@
+import { Activity, ActivityOptions } from "./activity";
+import { defaultLogger, Logger, YagnaApi } from "../utils";
+import { AgreementPoolService } from "../agreement";
+import { PaymentService } from "../payment";
+
+interface ActivityServiceOptions extends ActivityOptions {}
+
+/**
+ * Activity Pool Service
+ * A very simple implementation of the Activity Pool Service that allows to retrieve Activity from the pool.
+ * If the pool is empty, a new activity is created using the agreement provided by AgreementPoolService.
+ * @hidden
+ */
+export class ActivityPoolService {
+  private logger: Logger;
+  private pool: Activity[] = [];
+  private isServiceRunning = false;
+  constructor(
+    private yagnaApi: YagnaApi,
+    private agreementService: AgreementPoolService,
+    private paymentService: PaymentService,
+    private options?: ActivityServiceOptions,
+  ) {
+    this.logger = this.logger = options?.logger || defaultLogger();
+  }
+
+  /**
+   * Start ActivityPoolService
+   */
+  async run() {
+    this.isServiceRunning = true;
+    this.logger.debug("Activity Pool Service has started");
+  }
+
+  /**
+   * Get an activity from the pool of available ones or create a new one
+   */
+  async getActivity(): Promise<Activity> {
+    if (!this.isServiceRunning) {
+      throw new Error("Unable to get activity. Activity service is not running");
+    }
+    return this.pool.shift() || (await this.createActivity());
+  }
+
+  /**
+   * Release the activity back into the pool or if it is not reusable
+   * it will be terminated and the agreement will be released
+   * @param activity
+   * @param allowReuse
+   */
+  async releaseActivity(activity: Activity, allowReuse: boolean = true) {
```

Comment:
I think `activityPoolService.releaseActivity(activity, true)` is confusing to the reader. It should be obvious what each parameter does and just `true` doesn't tell me anything. IMO the last argument should be refactored to take an object `activityPoolService.releaseActivity(activity, { allowReuse: true})`

Reply:
also if it's always reused when `true` then maybe `resuse` is a better name than `allowReuse`